crates/uv/src/commands/pip/tree.rs

```diff
@@ -252,15 +258,25 @@ impl<'a> DisplayDependencyGraph<'a> {
     fn render(&self) -> Result<Vec<String>> {
         let mut visited: FxHashMap<PackageName, Vec<Requirement<VerbatimParsedUrl>>> =
             FxHashMap::default();
-        let mut path: Vec<PackageName> = Vec::new();
         let mut lines: Vec<String> = Vec::new();
 
-        // The starting nodes are the ones without incoming edges.
-        for site_package in self.site_packages.iter() {
-            // If the current package is not required by any other package, start the traversal
-            // with the current package as the root.
-            if !self.required_packages.contains(site_package.name()) {
-                lines.extend(self.visit(site_package, &mut visited, &mut path)?);
+        if self.package.is_empty() {
+            // The starting nodes are the ones without incoming edges.
+            for site_package in self.site_packages.iter() {
+                // If the current package is not required by any other package, start the traversal
+                // with the current package as the root.
+                if !self.required_packages.contains(site_package.name()) {
+                    lines.extend(self.visit(site_package, &mut visited, &mut Vec::new())?);
+                }
+            }
+        } else {
+            for (index, package) in self.package.iter().enumerate() {
+                if index != 0 {
+                    lines.push(String::new());
```

Comment:
Why are we adding newlines? We don't do this in the regular output.

Reply:
`cargo tree` seems to do this with `-p`, presumably so that it's clear each output block corresponds to a different `--package` value.

```
$ cargo tree -p futures-util -p tracing
futures-util v0.3.30
├── futures-core v0.3.30
├── futures-task v0.3.30
├── pin-project-lite v0.2.14
└── pin-utils v0.1.0

tracing v0.1.40
├── log v0.4.21
├── pin-project-lite v0.2.14
└── tracing-core v0.1.32
    └── once_cell v1.19.0
```

Reply:
This seems sensible to me.